pkg/controller/federatedtypeconfig/controller.go

```diff
@@ -281,6 +293,6 @@ func (c *Controller) removeFinalizer(tc *corev1a1.FederatedTypeConfig) error {
 	}
 	finalizers.Delete(finalizer)
 	accessor.SetFinalizers(finalizers.List())
-	_, err = c.client.FederatedTypeConfigs(tc.Namespace).Update(tc)
+	_, err = c.client.FederatedTypeConfigs(tc.Namespace).UpdateStatus(tc)
```

Comment:
ditto

Reply:
`UpdatesStatus()` method was used twice, one in the `ensureFinalizer` and other in `removeFinalizer`. so basically it was updating the status twice in the same iteration loop.  I commented out `UpdatesStatus()` in the `ensureFinalizer` method and it's working fine without any issues.